cpp/arcticdb/storage/azure/azure_mock_client.cpp

```diff
@@ -63,81 +52,62 @@ std::optional<Azure::Core::RequestFailedException> has_failure_trigger(const std
     }
 }
 
+Azure::Core::RequestFailedException MockAzureClient::missing_key_failure() const {
+    auto error_code = AzureErrorCode_to_string(AzureErrorCode::BlobNotFound);
+    std::string message = fmt::format("Simulated Error, message: Failed to find blob, {} {}", error_code,
+                                      (int) Azure::Core::Http::HttpStatusCode::NotFound);
```

Comment:
While we're here I agree with Phoebus' comment in the other PR that we should replace these C-style casts with `static_cast`

Reply:
Fixed in [3283f32](https://github.com/man-group/ArcticDB/pull/1395/commits/3283f32c0a323b10940abd25d0b876ca8b57984e)